janusgraph-backend-testutils/src/main/java/org/janusgraph/graphdb/JanusGraphTest.java

```diff
@@ -2959,17 +2960,31 @@ private void executeLockConflictingTransactionJobs(JanusGraph graph, Transaction
                 tx2.commit();
                 fail("Storage backend does not abort conflicting transactions");
             } catch (JanusGraphException ignored) {
+                Throwable rootCause = ignored.getCause().getCause();
+                assertTrue(rootCause instanceof PermanentLockingException);
+                assertTrue(rootCause.getMessage().contains("Expected value mismatch for"));
             }
         } else {
             try {
```

Comment:
It would cool to replace these lines with `assertThrows`.

Reply:
AFAIK, `assertThrows` can only assert the outmost exception but not inner exception(s).

Reply:
```
JanusGraphException janusGraphException = assertThrows(JanusGraphException.class, () -> tx2.commit());
Throwable rootCause = janusGraphException.getCause().getCause();
assertTrue(rootCause instanceof PermanentLockingException);
assertTrue(rootCause.getMessage().contains("Expected value mismatch for"));
```

Reply:
This was my idea.